openssl/src/asn1.rs

```diff
@@ -84,6 +84,29 @@ impl fmt::Display for Asn1GeneralizedTimeRef {
     }
 }
 
+impl Asn1GeneralizedTime {
+    /// Creates a new generalized time corresponding to the specified ASN1 time
+    /// string.
+    #[corresponds(ASN1_GENERALIZEDTIME_set_string)]
+    #[allow(clippy::should_implement_trait)]
+    pub fn from_str(s: &str) -> Result<Asn1GeneralizedTime, ErrorStack> {
+        unsafe {
+            ffi::init();
+
+            let time_str = CString::new(s).unwrap();
+            let ptr = cvt_p(ffi::ASN1_GENERALIZEDTIME_new())?;
+            let time = Asn1GeneralizedTime::from_ptr(ptr);
+
+            cvt(ffi::ASN1_GENERALIZEDTIME_set_string(
+                time.as_ptr(),
+                time_str.as_ptr(),
+            ))?;
+
+            Ok(time)
```

Comment:
Probably just a matter of taste, but I'd have written this as
```suggestion
            cvt(ffi::ASN1_GENERALIZEDTIME_set_string(ptr, time_str.as_ptr()))?;
            Ok(Asn1GeneralizedTime::from_ptr(ptr))
```

Reply:
Oh right. It's been too long... Thanks.